### ISSUE TITLE
Better story for undeleting term (after undo state is cleared)

### ISSUE THREAD

**Issue by ChrisPenner** [unisonweb/unison#2480]
`ucm version: release/M2j (built on 2021-10-07)`

I recently deleted a term, did some other things, then realized I needed to get it back/rename it.

* First I tried `reflog`, this shows the namespace related to the change, but not the hash of the term I deleted
* I tried `names` with the old name, wondering if it would find the old name
* I then tried `history` with the namespace's hash from reflog, but it also didn't show the hash of the deleted term:

```
.> history #oe82ov8sed
  Note: The most recent namespace hash is immediately below this message.

  ⊙ #oe82ov8sed

    - Deletes:

      base.test.durationPlus

```

* Next I tried diffing the namespace, surely it would then show the term which was deleted?

```
.> diff.namespace #oe82ov8sed

  I expected an identifier but found #oe82ov8sed
```

Looks like hashes are no good here;

* Next I tried to just undo a bunch; hoping `undo` would be scoped to the current branch, but unfortunately it wasn't.

* In the end I just kept redo-ing global state to get what I wanted, not ideal.
* Even after undoing to a good state, I then checked my history so that I could re-load the history state before undoing; but it looks like the undo-ing itself doesn't show up in the history (which makes sense I suppose, but I could also see it working like a git-style revert which makes a new change when undoing).

Long story short, should probably have a better user-experience here 😄

**Comment by pchiusano** [comment]
Yeah - one workaround is you can fork any namespace hash into your current tree, then you can diff it there.

Ideally diff could just work even if one or both namespaces are hashes. I think we have an issue to track that. If we did have that, you might just diff two entries in the reflog, then pass a numbered arg from the diff output to alias.term.
